build/matches-in-text.ts

```diff
@@ -1,35 +1,55 @@
+import fromMarkdown from "mdast-util-from-markdown";
+import visit from "unist-util-visit";
+
 const ESCAPE_CHARS_RE = /[.*+?^${}()|[\]\\]/g;
 
-export function* findMatchesInText(
-  needle,
-  haystack,
+type Match = { line: number; column: number };
+
+export function findMatchesInText(
```

Comment:
let this be a normal function, for we have convert all the results to an array in usage